open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/OpenMetadataAPIMapper.java

```diff
@@ -746,11 +746,23 @@
     public static final String NOTE_ENTRY_TYPE_NAME                     = "NoteEntry";              /* from Area 1 */
     /* Referenceable */
 
-    public static final String NOTE_LOG_AUTHOR_RELATIONSHIP_TYPE_GUID  = "8f798c0c-6769-4a2d-b489-12714d89e0a4";
+    /*
+    The below constants which contain the pattern
+    ..AUTH..="GUID"
+    looks exactly like
+    myAuth = SECURITY_SECRET
+    which is considered a security risk.
+    As both constants are currently unused I've commented them out assuming they are intended to be used in the future.
+    If the contsnats are re enabled in the future they will raise 2 HIGH Security Hotspot errors on SONAR cloud.
```

Comment:
If the code was actually being used then I agree we should just turn that rule off in this case.
However the fact that code wasn't being  used meant that I would like the option to track the "unused", rather than the original security alert , but I also wanted rid of the High sev alert.

Reply:
The security alert is clearly invalid in this case and needs to be flagged in the tool rather than making changes to the code.  This way the tool vendors improve their rules.

As new types are defined contributors are supposed to add the name and GUID into this mapping file so that they are ready for use in the generic handlers.  This is to increase the chances that the GUID values are right and discourage each OMAS from creating their own definitions. 

I am not sure it is helpful to flag the values that are not used in this list - it just creates more work when they are used (in fact they are likely to stay flagged as unused as they are used which is more misleading).  If it is a real problem then I woudl suggest putting one suppressWarnings at the top of the file.

Reply:
Thanks @mandy-chessell for responding ,
another issue with this code is that it's also wrongly being confused for a High Sev Security Hotspot.

A constant containing AUTH being assigned a Hex string , looks exactly like a developer has a hard code auth token in the source code. 

`public static final String NOTE_LOG_AUTHOR_RELATIONSHIP_TYPE_GUID  = "8f798c0c-6769-4a2d-b489-12714d89e0a4";
`
in this case that is wrong and could be turned off either locally or centrally and as I don't know how to change the the rules centrally , I've pushed a suggest fix that just turns off that specific alert. If you and Nigel find it acceptable I will open a new PR which turns off warnings for  only the 2 lines that are causing this alert to be raised, which  also happen to be our Highest Sev Security alerts in this scan.

Reply:
The tool is creating a false positive since clearly this is setting a variable to a string that is in no way related to security.  

Ideally we need to get the scanning tool fixed.  If we can configure the tool so that the false posititive is recorded and this is passed back to the developers of the tool then there is a chance they will fix it.  If we just turn off the scanning then it will come back when an external party runs their own scan.

If there is no way to configure the tool then disabiling the rule as you suggest is the best we can do.